src/linuxDistribution.ts

```diff
@@ -0,0 +1,67 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as fs from 'fs';
+import * as os from 'os';
+
+export class LinuxDistribution {
+    constructor(public name: string, public version: string) { }
+
+    /**
+     * There is no standard way on Linux to find the distribution name and version.
+     * Recently, systemd has pushed to standardize the os-release file. This has
+     * seen adoption in "recent" versions of all major distributions.
+     * https://www.freedesktop.org/software/systemd/man/os-release.html
+     */
+    public static GetDistroInformation(): Promise<LinuxDistribution> {
+        let linuxDistro: Promise<LinuxDistribution>;
+
+        // First check /etc/os-release and only fallback to /usr/lib/os-release
+        // as per the os-release documentation.
+        linuxDistro = LinuxDistribution.getDistroInformationFromFile('/etc/os-release')
+            .catch(() => {
+                return LinuxDistribution.getDistroInformationFromFile('/usr/lib/os-release');
+            }).catch(() => {
+                // couldn't get distro information
+                return Promise.resolve(new LinuxDistribution('unknown', 'unknown'));
+            });
+
+        return linuxDistro;
+    }
+
+    private static getDistroInformationFromFile(path: string): Promise<LinuxDistribution> {
+        return new Promise<LinuxDistribution>((resolve, reject) => {
+            fs.readFile(path, 'utf8', (error, data) => {
+                if (error) {
+                    reject(error);
+                    return;
+                }
+                resolve(LinuxDistribution.getDistroInformation(data));
+            });
+        });
+    }
+
+    // Only public for tests.
+    public static getDistroInformation(data: string): LinuxDistribution {
```

Comment:
I understand why this is here, but still... I find it a bit weird to have a _public_ method, with the **same name** (except the case of the first char) as another one, just for tests... This might lead to confusion down the road.

If this really needs to remain public, but you want to make it extra clear this is _not_ to be used, how about prefixing it with an `_` (underscore), the old-school way?